checkpoint_engine/ps.py

```diff
@@ -1262,7 +1264,19 @@ def _update_per_bucket(
                         buffer_b.data.copy_(h2d_buffer[: bucket.size])
                 brank = bcast_rank_map[receiver_rank]
                 dist.broadcast(buffer_b, src=brank)
-                socket.recv()
+                resp = socket.recv()
+                ret_code = torch.tensor(0, device="cuda")
+                if resp != b"":
+                    logger.error(
+                        f"[rank{self._rank}] receive error response from rank {receiver_rank} for bucket {gidx} in checkpoint {checkpoint_name}"
+                    )
+                    ret_code.fill_(1)
+                dist.all_reduce(ret_code, op=dist.ReduceOp.SUM)
+                torch.cuda.synchronize()
+                if ret_code.item() != 0:
+                    # quit early if any rank failed
+                    socket.send_pyobj(None)
```

Comment:
This will trigger process_weight_after_update, I think this should send a exception and vLLM worker should check it.